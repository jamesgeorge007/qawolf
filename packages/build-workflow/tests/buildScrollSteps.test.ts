import { loadEvents } from "@qawolf/fixtures";
import { buildScrollSteps } from "../";

describe("buildScrollSteps", () => {
  let scrollGoogle: any[];

  beforeAll(async () => {
    scrollGoogle = await loadEvents("google");
  });

  it("builds correct steps for scroll google", async () => {
    const steps = buildScrollSteps(scrollGoogle);
    expect(steps).toMatchSnapshot();
  });
});
