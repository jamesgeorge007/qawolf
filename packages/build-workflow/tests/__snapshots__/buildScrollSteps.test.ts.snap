// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildScrollSteps builds correct steps for scroll google 1`] = `
Array [
  Object {
    "action": "scroll",
    "index": 121,
    "pageId": 0,
    "target": Object {
      "xpath": "/html",
    },
    "value": Object {
      "x": 0,
      "y": 718,
    },
  },
  Object {
    "action": "scroll",
    "index": 202,
    "pageId": 0,
    "target": Object {
      "xpath": "/html",
    },
    "value": Object {
      "x": 0,
      "y": 780,
    },
  },
]
`;
